packages/monaco/src/browser/monaco-status-bar-contribution.ts

```diff
@@ -0,0 +1,76 @@
+ */
+
+import { injectable, inject } from 'inversify';
+import { DisposableCollection } from '@theia/core';
+import { FrontendApplicationContribution, FrontendApplication, StatusBar, StatusBarAlignment } from "@theia/core/lib/browser";
+import { EditorCommands, EditorManager, EditorWidget } from '@theia/editor/lib/browser';
+import { OutlineViewService } from '@theia/outline-view/lib/browser/outline-view-service';
+import { MonacoEditor } from './monaco-editor';
+
+@injectable()
+export class MonacoStatusBarContribution implements FrontendApplicationContribution {
+
+    protected readonly toDispose = new DisposableCollection();
+
+    constructor(
+        @inject(OutlineViewService) protected readonly outlineViewService: OutlineViewService,
+        @inject(EditorManager) protected readonly editorManager: EditorManager,
+        @inject(StatusBar) protected readonly statusBar: StatusBar
+    ) { }
+
+    onStart(app: FrontendApplication): void {
+        this.statusBar.onStatusBarEntriesLoaded(() => {
+            this.removeConfigTabSizeWidget();
+        });
+
+        this.editorManager.onCurrentEditorChanged(widget => {
+            if (widget) {
+                this.setConfigTabSizeWidget();
+
+                const editor = this.editorManager.currentEditor;
+                if (editor && this.getModel(editor)) {
+                    const editorModel = this.getModel(editor);
+                    if (editorModel) {
+                        this.toDispose.dispose();
+                        this.toDispose.push(editorModel.onDidChangeOptions(() => {
+                            this.setConfigTabSizeWidget();
+                        }));
+                    }
+                }
+            } else {
+                this.removeConfigTabSizeWidget();
+            }
+        });
+    }
+
+    protected setConfigTabSizeWidget() {
+        const editor = this.editorManager.currentEditor;
+        if (editor && this.getModel(editor)) {
+            const editorModel = this.getModel(editor);
+            if (editorModel) {
+                const modelOptions = editorModel.getOptions();
+                const tabSize = modelOptions.tabSize;
+                const useSpaceOrTab = modelOptions.insertSpaces ? 'Spaces' : 'Tabs';
+                this.statusBar.setElement('editor-status-tabbing-config', {
+                    text: `${useSpaceOrTab}: ${tabSize}`,
+                    alignment: StatusBarAlignment.RIGHT,
+                    priority: 10,
+                    command: EditorCommands.CONFIG_TAB_SIZE.id
+                });
+            }
+        }
+    }
+    protected removeConfigTabSizeWidget() {
+        this.statusBar.removeElement('editor-status-tabbing-config');
+    }
+
+    protected getModel(editor: EditorWidget): monaco.editor.IModel {
+        const monacoEditor = MonacoEditor.get(editor);
+        return monacoEditor!.getControl().getModel();
```

Comment:
`return monacoEditor && monacoEditor.getControl().getModel()`;

Reply:
sorry i do not understand why `monacoEditor && monacoEditor.getControl().getModel()` would be an instance of  `monaco.editor.IModel | undefined`
maybe you want me to do 
`
protected getModel(editor: EditorWidget): monaco.editor.IModel | undefined {
   const monacoEditor = MonacoEditor.get(editor);
   if (!monacoEditor) { 
     return undefined;
   }
   return manocoEditor.getControl().getModel();
}`
instead ?

Reply:
It is fine.

fyi: in JS evaluating logical operators to boolean value is a special case, i.e. `expr1 && expr2` is evaluated like `Returns expr1 if it can be converted to false; otherwise, returns expr2.` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators

so if `monacoEditor` is `undefined` and it can be converted to `false` then `undefined` is returned; otherwise a value of `monacoEditor.getControl().getModel()`